diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp

```diff
@@ -375,10 +372,12 @@ class Updater : public DiagnosticTaskVector
    * parameter.
    */
   Updater(
-    ros::NodeHandle h = ros::NodeHandle(), ros::NodeHandle ph = ros::NodeHandle("~"),
-    std::string node_name = ros::this_node::getName())
+    rclcpp::Node::SharedPtr h = rclcpp::Node::make_shared("diagnostics_updater"),
+    rclcpp::Node::SharedPtr ph = rclcpp::Node::make_shared("diagnostics_updater_private"),
```

Comment:
Does it really make sense to create two new ROS nodes here? I'm not too familiar with ROS2, but I think this actually creates a new node.